spec/features/discounts/cart_spec.rb

```diff
@@ -171,5 +175,41 @@
 
       expect(page).to have_content("Total: $240.00")
     end
+
+    it "discounts do not apply to items in the cart not from that merchant" do
+      visit "/items/#{@paper.id}"
+      click_on "Add To Cart"
+      visit "/items/#{@bluepill.id}"
+      click_on "Add To Cart"
+
+      visit "/cart"
+
+      within("#cart-item-#{@bluepill.id}") do
+        click_on "Increase Amount"
+        click_on "Increase Amount"
+        click_on "Increase Amount"
+
+        click_on "Increase Amount"
+        click_on "Increase Amount"
+
+        click_on "Increase Amount"
+        click_on "Increase Amount"
+
+        click_on "Increase Amount"
+        click_on "Increase Amount"
+      end
+
+
+      within "#cart-item-#{@bluepill.id}" do
+        expect(page).to have_link(@bluepill.name)
+        expect(page).to have_css("img[src*='#{@bluepill.image}']")
+        expect(page).to have_link("#{@bluepill.merchant.name}")
+        expect(page).to have_content("$70.00")
+        expect(page).to have_content("10")
+        expect(page).to have_content("$700.00")
+
+        expect(page).to_not have_content("Item price has been adjusted due to discount!")
```

Comment:
Makes sure even though there is 10 of this item, it is not of the merchant with the discount so the discount doesn't apply.